### ISSUE TITLE
Keypad nums recognized as num keys

### ISSUE THREAD

**Issue by Mirarchi** [Julow/Unexpected-Keyboard#723]
While sending the numeric values in keypad, i would expect to receive a specific keypad keycode (keycode.numpad0, keycode.numpad1 and so on) instead of receiving num keys (Keycode.num1, Keycode.num2 and so on) .
This makes impossible making custom character combinations (for example leftAlt+0123 for '{' character).
Any clue on this?

Thanks

**Comment by Julow** [comment]
Would Android recognize such sequences ? Have you tried with an other virtual virtual keyboard or a physical one ?

Related issue about sending key events: https://github.com/Julow/Unexpected-Keyboard/issues/717
Related issue about alt codes: https://github.com/Julow/Unexpected-Keyboard/issues/191 https://github.com/Julow/Unexpected-Keyboard/issues/686

**Comment by Mirarchi** [comment]
Yes, android ( at least in version 11) recognizes those sequences. Already tested with another keyboard (such as Hacker Keyboard) and this works.
Thanks.

**Comment by Julow** [comment]
That doesn't immediately solve your issue but with 2060e6a you can now add keypad buttons to your custom layouts:

```
:keyevent symbol='0':144
```

Key event numbers for keypad numbers are here: https://developer.android.com/reference/android/view/KeyEvent#KEYCODE_NUMPAD_0
